### PR TITLE
[backend] feature(freetrials/chunk4): endpoint to retrieve deployment requests #1135

### PR DESCRIPTION
**Pull request by carinelebas**
# Context
Create a new endpoint for platform to retrieve the deployment requests

## How to test

Use the graphql-api to:
- create deployment requests with different parameters.
- retrieve them using the filter

## Related issues

- Related to #1135

# Checklist

## Quality

- [x] I consider this work complete and ready for review
- [x] I tested all features and edge cases
- [x] I refactored code where necessary to improve quality
- [x] I made sure my code meets development guidelines
- [x] I performed a self-review of my code

## Testing

- [x] I wrote test cases for the relevant use cases (unit, integration or e2e tests)
- [ ] (Bug fixes only) I added tests that reproduce and verify the fix

### What tests have been made

- [x] Unit tests
- [ ] Integration tests
- [ ] E2E tests
- [x] Local manual tests

# Additional information
> Any additional context, dependencies, or concerns reviewers should know about.

### CODE REVIEW

File: apps/portal-api/src/modules/services/deployments/deployments.graphql [hunk 1]
```diff
@@ -40,6 +59,37 @@ input CreateDeploymentRequestInput {
...
+type DeploymentRequestConnection {
+  totalCount: Int!
+  edges: [DeploymentRequestEdge!]!
+  pageInfo: PageInfo!
+}
+
+extend type Query {
+  deploymentRequests(
+    first: Int!
+    after: ID
```
[the remark this answers is not in the document]
Comment (carinelebas) [reply apps/portal-api/src/modules/services/deployments/deployments.graphql]: [UserOrdering](https://github.com/FiligranHQ/xtm-hub/blob/f2242b6b726a75856fcdc493da5be3610e8193e2/apps/portal-api/src/__generated__/resolvers-types.ts#L1652-L1659) is liked to User data. In this case, we don't have the userData, so it would mean create another appropriate ordering. Since this is not something we'll need for now, I think it's best to keep things simple. Wdyt?

File: apps/portal-api/src/modules/services/deployments/deployments.app.test.ts [hunk 2]
```diff
@@ -53,6 +67,132 @@ describe('Deployment app', () => {
...
+
+      await DeploymentRequestDomain.deleteDeploymentRequestBy({
+        id: deploymentRequest1?.id,
+      });
+      await DeploymentRequestDomain.deleteDeploymentRequestBy({
+        id: deploymentRequest2?.id,
+      });
+      await DeploymentRequestDomain.deleteDeploymentRequestBy({
+        id: deploymentRequest3?.id,
+      });
```
[the remark this answers is not in the document]
Comment (carinelebas) [reply apps/portal-api/src/modules/services/deployments/deployments.app.test.ts]: good idea, it simplifies things ! done in [86386d1](https://github.com/FiligranHQ/xtm-hub/pull/1153/commits/86386d111f4ad5b47e7da4cbc4e471aede59cf3c)

File: apps/portal-api/src/modules/services/deployments/deployments.app.test.ts [hunk 3]
```diff
@@ -53,6 +67,132 @@ describe('Deployment app', () => {
...
+        id: deployment?.id as DeploymentRequestId,
+      });
+    });
+  });
+  describe('loadDeploymentRequests', () => {
+    async function insertOpenCtiDeploymentRequest(
+      deploymentRequest: Partial<DeploymentRequestInitializer>
+    ) {
+      const serviceInstanceId = uuidv4();
+      await db('ServiceInstance').insert([
```
[the remark this answers is not in the document]
Comment (carinelebas) [reply apps/portal-api/src/modules/services/deployments/deployments.app.test.ts]: good idea, done in [86386d1](https://github.com/FiligranHQ/xtm-hub/pull/1153/commits/86386d111f4ad5b47e7da4cbc4e471aede59cf3c)
